### DISCUSSION TITLE
Where is the L0 and P(persistence length) inside the code?

### DISCUSSION THREAD

**Discussion by ErenC1** [UvaCsl/HemoCell#55] [General discussion]
Hi!

Actually I found my answer, apparently maxCellPersistence length is tau_link^2. So my remaining question is, where is the L0 and P(persistence length)? If I change RBC resolution I need to change parameters accordingly, and I couldn't find where they used inside the code.

Best,
Eren.

**Comment by gzavo** [comment]
Hi @ErenC1,

Everything related to mechanical models is in the 'mechanics' folder.

For RBCs, the link force uses persistence length i.e., 
https://github.com/UvaCsl/HemoCell/blob/cf106f34aa015ee28d8de32ccd440ff3fb5abb16/mechanics/rbcHighOrderModel.cpp#L181

And the limiting extent of the persistence length is defined here:
https://github.com/UvaCsl/HemoCell/blob/cf106f34aa015ee28d8de32ccd440ff3fb5abb16/config/constant_defaults.h#L173
Note that this is relative (3x length extension of the resting length at the moment), so it should work with different resolution as well.

**Comment by ErenC1** [comment]
Thank you @gzavo!

Actually what I am asking is this:

1) In publication (2.2-1) link force defined as:

F_link= k_link * dL/p * (1+ 1/(tau_link^2-dL^2)),

but in https://github.com/UvaCsl/HemoCell/blob/cf106f34aa015ee28d8de32ccd440ff3fb5abb16/mechanics/rbcHighOrderModel.cpp#L181 I see:

F_link= k_link * (dL+dL/tau_link^2-dL^2) = k_link * dL * (1+1/(tau_link^2-dL^2)

so, where is the p(given as 7.5 um in publication) division?

2) Same applies to the other forces, I can't see where they divided by L0...

3) In publication tau_volume is given by 0.01, but in code it looks like 0.1, and tau_volume^2 is 0.01. I tried both, tau_volume^2=0.01 gives better results (I finally validated the shear flow)... Any comment on this?

Thank you,
Eren.

**Comment by gzavo** [comment]
Hi @ErenC1,

Good questions!

For Q1 and Q2: since scaling with the numerical resolution has nothing to do with the actual mechanics of the cells, it is implemented outside of the specific cell mechanics. Look around here:
https://github.com/UvaCsl/HemoCell/blob/cf106f34aa015ee28d8de32ccd440ff3fb5abb16/mechanics/cellMechanics.h#L52

For Q3 it seems it has changed after the publication of the original paper. I need to check to be sure, but as I recall the main reason was that the overall effect on many cell flows (e.g., on viscosity, margination, etc.) was close to zero when allowing a bit more compressibility, but it improved numerical stability substantially for high shear rate flows, which is what we primarily investigate. This will indeed need to be incorporated in the documentation.

I hope this answers your questions!
Gabor

**Reply by ErenC1** [reply]
Hi @gzavo!

I actually checked that file too but distracted by "Kaniadakis magic" (ADHD problems...) but now it makes sense. 

I got all my answers, thank you!
Eren.
